### PR TITLE
fix: creation of CtTypeReferences from java classes

### PR DESCRIPTION
**Pull request by I-Al-Istannen**
Creating `CtTypeReference`s from java classes was a bit too ad-hoc for my usecase. This PR adds support for array types and anonymous classes. Anonymous classes do not appear in the shadow model but can still be useful when dealing with arbitrary objects. They are also supported by the source model.

Notable changes: This PR switches from `CLass#getDeclaringClass` to `Class#getEnclosingClass` to properly work with anonymous classes. This seems to be closer to what the source model uses as well.

### CODE REVIEW

**Comment by I-Al-Istannen** [comment]
@MartinWitt ping!

File: src/test/java/spoon/test/factory/CodeFactoryTest.java [hunk 1]
```diff
@@ -66,4 +68,54 @@ void createCtCatcVariablehWithoutModifiers() {
...
+		Class<?> clazz = (new Object() {
+		}.getClass());
+		Factory factory = createFactory();
+		assertEquals(
+			clazz.getName(),
+			factory.createCtTypeReference(clazz).getQualifiedName()
+		);
+	}
+
+	private static class TestClassInnerClass {}
```
Comment (MartinWitt) [inline src/test/java/spoon/test/factory/CodeFactoryTest.java:120]: Maybe move this test object closer to the usage?

Comment (I-Al-Istannen) [reply src/test/java/spoon/test/factory/CodeFactoryTest.java]: I inlined it.

**Review by MartinWitt — APPROVED** [review]

**Comment by MartinWitt** [comment]
Thanks @I-Al-Istannen

**Pull request merged**